src/net/tcp/tcp.cpp

```diff
@@ -65,49 +88,76 @@ TCP::TCP(IPStack& inet) :
   Current solution:
   Simple.
 */
-Listener& TCP::bind(const port_t port, ConnectCallback cb)
+Listener& TCP::listen(tcp::Socket socket, ConnectCallback cb)
 {
-  Listeners::const_iterator it = listeners_.find(port);
-  // Already a listening socket.
-  if(it != listeners_.cend()) {
-    throw TCPException{"Port is already taken."};
-  }
-  auto& listener = listeners_.emplace(port,
-    std::make_unique<tcp::Listener>(*this, port, std::move(cb))
+  bind(socket);
```

Comment:
What will happen on `TCP::listen({{8, 8, 8, 8}, 80}`? I think bind to socket has to verify the source IP with inet.